tests/eosio.system_tests.cpp

```diff
@@ -6026,4 +6026,218 @@ BOOST_FIXTURE_TEST_CASE( buy_pin_sell_ram, eosio_system_tester ) try {
 
 } FC_LOG_AND_RETHROW()
 
+
+BOOST_AUTO_TEST_SUITE_END()
+
+BOOST_AUTO_TEST_SUITE(eosio_system_account_name_restriction)
+
+// -----------------------------------------------------------------------------------------
+//             tests for account name restrictions (blacklist patterns)
+// -----------------------------------------------------------------------------------------
+
+template<class T>
+concept VectorLike = requires(T v) { v.begin(); v.end(); (void)v[0]; };
+
+template<VectorLike... Vs>  // returns a new vector which is the concatenation of the vectors passed as arguments
+auto cat(Vs&&... vs) {
+    std::common_type_t<Vs...> res;
+    res.reserve((0 + ... + vs.size()));
+    (..., (res.insert(res.end(), std::begin(std::forward<Vs>(vs)), std::end(std::forward<Vs>(vs)))));
+    return res;
+}
+
+BOOST_FIXTURE_TEST_CASE( restrictions_update, eosio_system_tester ) try {
+   const std::vector<account_name> accounts = { "alice"_n };
+   create_accounts_with_resources( accounts );
+   const account_name alice = accounts[0];
+
+   // make sure `denynames` requires "eosio"_n auth
+   // ----------------------------------------------
+   BOOST_REQUIRE_EQUAL(denynames(alice, {"bob"_n, "bob.yxz"_n}), error("missing authority of eosio"));
+
+   // check that `denynames` and `undenynames` update the blacklist as expected
+   // ------------------------------------------------------------------------
+   BOOST_REQUIRE_EQUAL(denynames("eosio"_n, {}), success());                // empty list is silently ignored.
+   
+   std::vector<name> add1 {"bob"_n, "bob.yxz"_n};
+   BOOST_REQUIRE_EQUAL(denynames("eosio"_n, add1), success());
+   BOOST_REQUIRE(get_blacklisted_names() == add1);                          // initial add works
+
+   std::vector<name> add2 {"alice.xyz.x"_n, "alice"_n};
+   BOOST_REQUIRE_EQUAL(denynames("eosio"_n, add2), success());              // appending works
+   BOOST_REQUIRE(get_blacklisted_names() == cat(add1, add2));
+
+   std::vector<name> add3 {"bob.yxz"_n, "alice"_n};
+   BOOST_REQUIRE_EQUAL(denynames("eosio"_n, add3), success());              // duplicates are ignored
+   BOOST_REQUIRE(get_blacklisted_names() == cat(add1, add2));
+
+   std::vector<name> add4 {"fred.xyz.x"_n, "fred"_n};
+   BOOST_REQUIRE_EQUAL(denynames("eosio"_n, cat(add4, add4, add4)), success()); // duplicates are ignored even within one call
+   BOOST_REQUIRE(get_blacklisted_names() == cat(add1, add2, add4));
+
+   std::vector<name> add5 {""_n, "alice1alice2x"_n};
+   BOOST_REQUIRE_EQUAL(denynames("eosio"_n, add5), success());             // invalid names are ignored
```

Comment:
`alice1alice2x` isn't even a valid name since the 13th character is too large. Maybe make the last character `a` or something. Also now that bad patterns would not just be silently ignored, I would expect to see this fail with an appropriate error message.

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/d526c1695c932b71f7e6f517d962ebdcd482bf6c

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/8aaef04c6c09baca2490c02dd9b08ae0c75df0aa